### ISSUE TITLE
Abnormal computation time when carrying out experiment?

### ISSUE THREAD

**Issue by jsvamih** [saurabhdash/GCN_Partitioning#5]
I tested with my real data which has ~300 nodes and 8000 edges. It takes more than 1 day to perform the training. (Also tested data with 40 nodes and takes 10s every epoch. ) While the paper shows a computation time 1000ms for 1k nodes.  Any probable reason from configuration? I used python 3.8.5,  cuda 11.1, pytorch1.8.1 and other dependencies same as the requirements.txt.  I used one Quadro P5000 graphic.
On the other hand, are there any computation time result with  larger data provided for reference?
Thanks a lot.

**Comment by saurabhdash** [comment]
The Custom Loss involves a for loop in python which is extremely inefficient. Since the adjacency matrix is small, you can use the default PyTorch backprop for the loss.
